webapp/channels/src/components/user_settings/display/render_emoticons_as_emoji/render_emoticons_as_emoji.tsx

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useState, useRef, useEffect} from 'react';
+import type {ReactNode} from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import type {PreferencesType, PreferenceType} from '@mattermost/types/preferences';
+
+import {Preferences} from 'mattermost-redux/constants';
+
+import SettingItemMax from 'components/setting_item_max';
+import SettingItemMin from 'components/setting_item_min';
+import type SettingItemMinComponent from 'components/setting_item_min';
+
+import {a11yFocus} from 'utils/utils';
+
+export type OwnProps = {
+    adminMode?: boolean;
+    userId: string;
+    userPreferences?: PreferencesType;
+}
+
+type Props = OwnProps & {
+    active: boolean;
+    areAllSectionsInactive: boolean;
+    renderEmoticonsAsEmoji: string;
+    updateSection: (section: string) => void;
+    renderOnOffLabel: (label: string) => ReactNode;
+    actions: {
+        savePreferences: (userId: string, preferences: PreferenceType[]) => void;
+    };
+}
+
+const RenderEmoticonsAsEmoji: React.FC<Props> = ({userId, active, areAllSectionsInactive, renderEmoticonsAsEmoji, updateSection, renderOnOffLabel, actions}) => {
+    const [renderEmoticonsAsEmojiState, setRenderEmoticonsAsEmojiState] = useState<string>(renderEmoticonsAsEmoji);
+
+    const [isSaving] = useState<boolean>(false);
+
+    const [serverError] = useState<string | undefined>(undefined);
+
+    const minRef = useRef<SettingItemMinComponent>(null);
+
+    const focusEditButton = () => {
+        minRef.current?.focus();
+    };
+
+    useEffect(() => {
+        if (active && areAllSectionsInactive) {
```

Comment:
I don't think this is correct since these can never be true at the same time when these are set this way:
```
                <RenderEmoticonsAsEmoji
                    active={this.props.activeSection === 'renderEmoticonsAsEmoji'}
                    areAllSectionsInactive={this.props.activeSection === ''}
```
Perhaps this should be `!active && areAllSectionsInactive`?

Reply:
Makes sense. I've changed it to `!active && areAllSectionsInactive`.